### PR TITLE
Switches from erb to haml

### PR DESCRIPTION
**Pull request by ashfurrow**
Had the sudden urge to try using [haml](http://haml.info), a different template engine from erb. I like that I can't forget to close tags.

Benefits:

- Can easily escape to markdown, erb, whatever else I need to for convenience. 
- Forced me to break out large amounts of ruby into their own files.
- I really like this.

Drawbacks:

- Haml is less familiar, may impede me down the line where erb wouldn't.
- Less approaching if I'm trying to show someone how I've do something.

This PR also cleans up a lot of other things, oops. 

Going to sleep on it. It's a big change, so some thorough testing should be done.

### CODE REVIEW

**Comment by ashfurrow** [comment]
Latest is on staging: http://staging.ashfurrow.com

**Comment by ashfurrow** [comment]
Now valid HTML5 :tada:

**Comment by orta** [comment]
I'm a fan of this level of abstraction: https://github.com/CocoaPods/cocoapods.org/blob/master/views/about.slim

File: lib/5-years-of-ios.rb [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    eos
+  end
+end
+
+module FiveYears
+
+  public
+
+  def self.Events
+    events = []
```
Comment (orta) [inline lib/5-years-of-ios.rb]: this stuff should ideally go in a middleman data file: https://middlemanapp.com/advanced/data_files/

**Comment by ashfurrow** [comment]
Neat! I picked Haml just because it seemed like the most popular – I was able to find answers to questions easily.

File: lib/5-years-of-ios.rb (hunk 1 above)
Comment (ashfurrow) [inline lib/5-years-of-ios.rb]: Yeah, loading form data would be a good idea.

**Comment by orta** [comment]
Yeah, slim is basically same-same but different

**Comment by ashfurrow** [comment]
Now that I _can_ write everything in markdown, I probably should.

**Pull request merged**